src/components/Calculator.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState } from "react";
+import { calculateOperations } from "../utils/claculator"; // Import the function
+
+type Results = {
+  adding: number;
+  multiplication: number;
+  division: number;
+  subtraction: number;
+};
+
+const Calculator: React.FC = () => {
+  const [num1, setNum1] = useState<number | null>(null);
+  const [num2, setNum2] = useState<number | null>(null);
+  const [results, setResults] = useState<Results | null>(null);
+  const [error, setError] = useState<string>("");
+
+  const handleCalculate = () => {
+    if (num1 === null || num2 === null) {
+      setError("Please enter valid numbers");
+      return;
+    }
+
+    if (num1 < 1 || num1 > 50) {
```

Comment:
Refactor the input validation for `num1` and `num2` into a separate function to reduce code duplication and improve maintainability. This function can take a number and its range as parameters and return any error messages. [important]